### ISSUE TITLE
KeyError: 'video_url' with videos within a sidecar post

### ISSUE THREAD

**Issue by saajo-lab** [instaloader/instaloader#558]
Some videos in profiles are causing KeyError. Hashtags are not affected (yet).

Versions from 4.2.10 to 4.3 are causing the same error.

For example:
`instaloader arianagrande`

Is causing this error:
```
Stored ID 7719696 for profile arianagrande.
arianagrande/2020-03-18_20-18-01_UTC_profile_pic.jpg 
Retrieving posts from profile arianagrande.
[  1/4532] arianagrande/2020-04-01_18-52-39_UTC.jpg [happy birthday to one of my f…] arianagrande/2020-04-01_18-52-39_UTC.mp4 json 
[  2/4532] arianagrande/2020-03-31_21-38-22_UTC_1.jpg arianagrande/2020-03-31_21-38-22_UTC_2.jpg arianagrande/2020-03-31_21-38-22_UTC_3.jpg arianagrande/2020-03-31_21-38-22_UTC_4.jpg arianagrande/2020-03-31_21-38-22_UTC_5.jpg arianagrande/2020-03-31_21-38-22_UTC_6.jpg arianagrande/2020-03-31_21-38-22_UTC_7.jpg arianagrande/2020-03-31_21-38-22_UTC_8.jpg arianagrande/2020-03-31_21-38-22_UTC_9.jpg arianagrande/2020-03-31_21-38-22_UTC_10.jpg [imy] json 
[  3/4532] arianagrande/2020-03-29_19-56-58_UTC.jpg [court knee] json 
[  4/4532] arianagrande/2020-03-29_18-45-41_UTC_1.jpg arianagrande/2020-03-29_18-45-41_UTC_2.jpg arianagrande/2020-03-29_18-45-41_UTC_3.jpg Warning: Use --login to download HD version of profile pictures.
Traceback (most recent call last):
  File "/usr/local/bin/instaloader", line 11, in <module>
    sys.exit(main())
  File "/home/pwah/.local/lib/python3.5/site-packages/instaloader/__main__.py", line 426, in main
    storyitem_filter_str=args.storyitem_filter)
  File "/home/pwah/.local/lib/python3.5/site-packages/instaloader/__main__.py", line 195, in _main
    fast_update, post_filter, storyitem_filter)
  File "/home/pwah/.local/lib/python3.5/site-packages/instaloader/instaloader.py", line 1106, in download_profiles
    downloaded = self.download_post(post, target=profile_name)
  File "/home/pwah/.local/lib/python3.5/site-packages/instaloader/instaloader.py", line 492, in download_post
    for sidecar_node in post.get_sidecar_nodes():
  File "/home/pwah/.local/lib/python3.5/site-packages/instaloader/structures.py", line 231, in get_sidecar_nodes
    video_url=node['video_url'] if is_video else None)
KeyError: 'video_url'
```

**Comment by mattgillooly** [comment]
I think it's happening when a given post contains a mixture of both photos and videos.

**Comment by mattgillooly** [comment]
I don't get the error if I skip such posts using the `--post-filter="typename != 'GraphSidecar'"` argument.  Perhaps that's helpful as a temporary measure until the bug is fixed.

**Comment by DReaper** [comment]
Can confirm I get the same error.

Retrieving IGTV videos for profile onepixcel_sub.
Retrieving posts from profile onepixcel_sub.
[  1/ 31] Traceback (most recent call last):
  File "__main__.py", line 459, in <module>
  File "__main__.py", line 455, in __main
  File "__main__.py", line 412, in main
  File "__main__.py", line 192, in _main
  File "instaloader\instaloader.py", line 1106, in download_profiles
  File "instaloader\instaloader.py", line 492, in download_post
  File "instaloader\structures.py", line 231, in get_sidecar_nodes
KeyError: 'video_url'
[25304] Failed to execute script __main__

**Comment by rathD25** [comment]
It's also happening to me

Retrieving posts from profile exampleprofile.
[  1/3672] Traceback (most recent call last):
  File "/home/user/.local/bin/instaloader", line 11, in <module>
    sys.exit(main())
  File "/home/user/.local/lib/python3.6/site-packages/instaloader/__main__.py", line 426, in main
    storyitem_filter_str=args.storyitem_filter)
  File "/home/user/.local/lib/python3.6/site-packages/instaloader/__main__.py", line 195, in _main
    fast_update, post_filter, storyitem_filter)
  File "/home/user/.local/lib/python3.6/site-packages/instaloader/instaloader.py", line 1106, in download_profiles
    downloaded = self.download_post(post, target=profile_name)
  File "/home/user/.local/lib/python3.6/site-packages/instaloader/instaloader.py", line 492, in download_post
    for sidecar_node in post.get_sidecar_nodes():
  File "/home/user/.local/lib/python3.6/site-packages/instaloader/structures.py", line 231, in get_sidecar_nodes
    video_url=node['video_url'] if is_video else None)
KeyError: 'video_url'

**Comment by gagharutyunyan1993** [comment]
**Same problem!**

Retrieving posts from profile alissaviolet.
[  1/641] alissaviolet/2020-04-01_00-23-50_UTC_1.jpg alissaviolet/2020-04-01_00-23-50_UTC_2.jpg alissaviolet/2020-04-01_00-23-50_UTC_3.jpg alissaviolet/2020-04-01_00-23-50_UTC_4.jpg alissaviolet/2020-04-01_00-23-50_UTC_5.jpg [quarantine land is so boring] json
[  2/641] alissaviolet/2020-03-12_17-25-37_UTC_1.jpg alissaviolet/2020-03-12_17-25-37_UTC_2.jpg [good doggie] json
[  3/641] alissaviolet/2020-03-10_22-17-29_UTC_1.jpg alissaviolet/2020-03-10_22-17-29_UTC_2.jpg alissaviolet/2020-03-10_22-17-29_UTC_3.jpg Traceback (most recent call last):
  File "C:\Users\gboy1\AppData\Local\Programs\Python\Python38-32\Scripts\instaloader-script.py", line 11, in <module>
    load_entry_point('instaloader==4.3', 'console_scripts', 'instaloader')()
  File "c:\users\gboy1\appdata\local\programs\python\python38-32\lib\site-packages\instaloader\__main__.py", line 412, in main
    _main(loader,
  File "c:\users\gboy1\appdata\local\programs\python\python38-32\lib\site-packages\instaloader\__main__.py", line 192, in _main
    instaloader.download_profiles(profiles,
  File "c:\users\gboy1\appdata\local\programs\python\python38-32\lib\site-packages\instaloader\instaloader.py", line 1106, in download_profiles
    downloaded = self.download_post(post, target=profile_name)
  File "c:\users\gboy1\appdata\local\programs\python\python38-32\lib\site-packages\instaloader\instaloader.py", line 492, in download_post
    for sidecar_node in post.get_sidecar_nodes():
  File "c:\users\gboy1\appdata\local\programs\python\python38-32\lib\site-packages\instaloader\structures.py", line 231, in get_sidecar_nodes
    video_url=node['video_url'] if is_video else None)
KeyError: 'video_url'

**Comment by Yakabuff** [comment]
same here

**Comment by aandergr** [comment]
Thanks a lot to all of you for reporting! We released [Version 4.3.1](https://github.com/instaloader/instaloader/releases/tag/v4.3.1) with a fix included.

Happy Instaloading and a nice weekend to you all!
